openedx/core/lib/api/view_utils.py

```diff
@@ -87,7 +87,8 @@ def _wrapper(self, request, *args, **kwargs):
                         request.user,
                         access_action,
                         course_id,
-                        depth=depth
+                        depth=depth,
+                        check_if_enrolled=True,
```

Comment:
*Note to reviewers:* Checking for enrollment was originally done in `can_load_mobile` in `access.py`.  But now, all enrollment checks are consolidated in `get_course_with_access`.